tests/integration/components/bs-form-test.js

```diff
@@ -159,6 +161,145 @@ module('Integration | Component | bs-form', function(hooks) {
     assert.dom(`.${formFeedbackClass()}`).hasText('There is an error');
   });
 
+  test('Yields #isSubmitting', async function(assert) {
+    this.set('submitAction', () => {
+      return new RSVP.Promise((resolve) => {
+        later(() => {
+          resolve();
+        }, 10);
+      });
+    });
+    await this.render(hbs`{{#bs-form onSubmit=submitAction as |form|}}
+      <div class='state {{if form.isSubmitting 'is-submitting'}}'></div>
+    {{/bs-form}}`);
+
+    assert.dom('form .state').doesNotHaveClass('is-submitting');
+
+    triggerEvent('form', 'submit');
+
+    await waitFor('.is-submitting');
+    assert.dom('form .state').hasClass('is-submitting');
```

Comment:
There are some test runs where this is failing, not sure why though!?

Reply:
I noticed that using timers to ensure not resolving in a too early runloop is not that reliable. I will refactor that tests to use `RSVP.defer()`. Should also improve readability.